lib/coverband/reporters/base.rb

```diff
@@ -61,16 +62,31 @@ def merge_arrays(first, second)
           merged
         end
 
+        ###
+        # This method is responsible for finding the CURRENT LOCAL
+        # filename regardless of the paths being different on
+        # various servers or deployments
+        ###
         def filename_from_key(key, roots)
-          filename = key
+          relative_filename = key
+          local_filename = relative_filename
           roots.each do |root|
-            filename = filename.gsub(/^#{root}/, './')
+            relative_filename = relative_filename.gsub(/^#{root}/, './')
           end
-          # the filename for  SimpleCov is expected to be a full path.
+          # the filename for our reports is expected to be a full path.
           # roots.last should be roots << current_root}/
           # a fully expanded path of config.root
-          filename = filename.gsub('./', roots.last)
-          filename
+          # filename = filename.gsub('./', roots.last)
+          # above only works for app files
+          # we need to rethink some of this logic
+          # gems aren't at project root and can have multiple locations
+          roots.each do |root|
+            if File.exist?(relative_filename.gsub('./', root))
+              local_filename = relative_filename.gsub('./', root)
+              break
+            end
+          end
```

Comment:
Not sure if this reads better using find to return the first root that has the file?
```ruby
  root = roots.find { |root| File.exist?(relative_filename.gsub('./', root)) }
  relative_filename.gsub('./', root)
```

Reply:
nice updated